<?xml version='1.0' encoding='utf-8'?>
<!--This does not replace URDF, and is not an extension of URDF.
    This is a format for representing semantic information about the robot structure.
    A URDF file must exist for this robot as well, where the joints and the links that are referenced are defined
-->
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="panda">
  <xacro:arg name="hand" default="true"/>
  <xacro:include filename="$(find panda_moveit_config)/config/panda.srdf.xacro"/>
  <!--GROUPS: Representation of a set of joints and links. This can be useful for specifying DOF to plan for, defining arms, end effectors, etc-->
  <!--LINKS: When a link is specified, the parent joint of that link (if it exists) is automatically included-->
  <!--JOINTS: When a joint is specified, the child link of that joint (which will always exist) is automatically included-->
  <!--CHAINS: When a chain is specified, all the links along the chain (including endpoints) are included in the group. Additionally, all the joints that are parents to included links are also included. This means that joints along the chain and the parent joint of the base link are included in the group-->
  <!--SUBGROUPS: Groups can also be formed by referencing to already defined group names-->
  <group name="panda_arm_hand">
    <group name="panda_manipulator"/>
    <group name="panda_hand"/>
  </group>
  <!--GROUP STATES: Purpose: Define a named state for a particular group, in terms of joint values. This is useful to define states like 'folded arms'-->
  <group_state name="ready" group="panda_arm">
    <joint name="panda_joint1" value="0" />
    <joint name="panda_joint2" value="-0.785" />
    <joint name="panda_joint3" value="0" />
    <joint name="panda_joint4" value="-2.356" />
    <joint name="panda_joint5" value="0" />
    <joint name="panda_joint6" value="1.571" />
    <joint name="panda_joint7" value="0.785" />
  </group_state>
  <group_state name="extended" group="panda_arm">
    <joint name="panda_joint1" value="0" />
    <joint name="panda_joint2" value="0" />
    <joint name="panda_joint3" value="0" />
    <joint name="panda_joint4" value="0" />
    <joint name="panda_joint5" value="0" />
    <joint name="panda_joint6" value="0" />
    <joint name="panda_joint7" value="0.785" />
  </group_state>
  <group_state name="open" group="panda_hand">
    <joint name="panda_finger_joint1" value="0.035" />
    <joint name="panda_finger_joint2" value="0.035" />
  </group_state>
  <group_state name="close" group="panda_hand">
    <joint name="panda_finger_joint1" value="0" />
    <joint name="panda_finger_joint2" value="0" />
  </group_state>
  <!--END EFFECTOR: Purpose: Represent information about an end effector.-->
  <end_effector name="panda_hand" parent_link="panda_link8" group="panda_hand" parent_group="panda_arm"/>
</robot>
